### PR TITLE
Fix nrf monotonics

### LINKED ISSUE #848: Race condition in `nrf:rtc` and `nrf:timer` monotonics
Same problem as #825, and will probably have the same solution.

### PR DESCRIPTION
**Pull request by Finomnis**
Fixes #848.

### CODE REVIEW

File: rtic-monotonics/src/nrf/timer.rs [hunk 1]
```diff
@@ -139,17 +140,42 @@ macro_rules! make_timer {
                 // 1 MHz
                 timer.prescaler.write(|w| unsafe { w.prescaler().bits(4) });
                 timer.bitmode.write(|w| w.bitmode()._32bit());
-                timer
-                    .intenset
-                    .modify(|_, w| w.compare0().set().compare1().set());
-                timer.cc[1].write(|w| unsafe { w.cc().bits(0) }); // Overflow
+
+                // Disable interrupts, as preparation
+                timer.intenclr.modify(|_, w| w
```
Comment (jamesmunns) [inline rtic-monotonics/src/nrf/timer.rs:145]: just checking, isn't intenclr "set to one to disable interrupts"? (if there's only a `clear` method that sets high, then ignore me :)

Comment (Finomnis) [reply rtic-monotonics/src/nrf/timer.rs:145]: I think yes, that's what clear does

File: rtic-monotonics/src/nrf/timer.rs [hunk 2]
```diff
@@ -139,17 +140,42 @@ macro_rules! make_timer {
...
+                timer.cc[2].write(|w| unsafe { w.cc().bits(0x8000_0000) }); // Half-period
+
+                // Reset the timer
                 timer.tasks_clear.write(|w| unsafe { w.bits(1) });
                 timer.tasks_start.write(|w| unsafe { w.bits(1) });
 
-                $tq.initialize(Self {});
-
+                // Clear pending events.
+                // Should be close enough to the timer reset that we don't miss any events.
```
Comment (Finomnis) [inline rtic-monotonics/src/nrf/timer.rs]: Might require a critical section to actually guarantee this.

**Review by korken89 — APPROVED** [review]
Thank you!

**Pull request merged**